src/scripts/community-toolbox.js

```diff
@@ -0,0 +1,266 @@
+      // If the stored data is not undefined or null, execution goes here
+      if(allContributors != null && allContributors!=undefined && allContributors.length>0) {
+
+        // Flushes repoContributors from the database after every single day
+        let timeNow = (new Date).getTime();
+        let lifespan = model_utils.getItem(`${repo}Expiry`);
+        if (lifespan!=null && ((timeNow-lifespan)/1000) >= 86400) {
+          console.log("deleting entries...");
+          model_utils.deleteItem(`${repo}`);
+          model_utils.deleteItem(`${repo}Expiry`);
+        }
+        
+        // Looking for repo Contributors list in the database
+        model_utils.getItem(repo).then((repoContributors) => {
+          // If we don't have repoContributors in the database, we fetch them from Github
+          if (repoContributors == null || repoContributors == undefined || repoContributors.length == 0) {
+            repoContributorsUtility.fetchRepoContributorsUtil(org, repo)
+            .then(function gotRepoContributorsInStorage (contributors) {
+              ui.insertContributors(contributors);
+              return;
+            })
+          }
+          // If we have repoContributors in the database, we save a network call :)
+          else {
+            ui.insertContributors(repoContributors);
+            return;
+          }
+        })
+      } else {
+        // Execution goes here, it means that data for this repo is not available
+        // in the database
+        console.log(`Something went wrong while getting ${repo} contributors :(`);
+      }
+    })
+  }
+
+  // Stores all the Recent Contributors in the database
+  function storeAllRecentContribsInitially(org, repo) {
+    // We make queryTime 1 month behind the current time, to pass it as query in the request
+    let d = (new Date);
+    d.setDate(d.getDate() - 30);
+    let queryTime = d.toISOString();
+    let repos = JSON.parse(localStorage.getItem('repos'));
+    return model_utils.getItem('recent-present').then((result)=> {
+      if(result!=null && result!=undefined) {
+        return result;
+      }
+      else {
+        if(repos!=null || repos!=undefined) {
+          return getRecentCommitsUtility.fetchAllRecentMonthCommits(repos, queryTime)
+                  .then((result) => {
+                    model_utils.setItem('recent-present', 'true');
+                    return result;
+                  })
+        }
+        else {
+          getAllContribsUtility.getAllRepos(org).then((repos) => {
+            if(repos!=null || repos!=undefined || repos.length>0) {
+              return getRecentCommitsUtility.fetchAllRecentMonthCommits(repos, queryTime)
+                      .then((result) => {
+                        model_utils.setItem('recent-present', 'true');
+                        return result;
+                      })
+            }
+          });
+        }
+      }
+    });
+  }
+
+
```

Comment:
This file is starting to get a bit long; if possible, we should break it up into subfiles along MVC lines!

Reply:
Yes, I also feel like so...I'll try to do it :+1: